src/main/org/pantsbuild/jarjar/misplaced/MisplacedClassProcessorFactory.java

```diff
@@ -0,0 +1,70 @@
+package org.pantsbuild.jarjar.misplaced;
+
+import org.pantsbuild.jarjar.util.ProcessingContext;
+
+/**
+ * Factory singleton for creating a MisplacedProcessorFactory from a user-level strategy string.
+ */
+public class MisplacedClassProcessorFactory {
+
+  /** Strategy which fails-fast with an exception upon hitting misplaced class. */
+  public static final String STRATEGY_FATAL = "fatal";
+
+  /** Strategy which skips past misplaced classes, excluding them from shading. */
+  public static final String STRATEGY_SKIP = "skip";
+
+  /** Strategy which omits misplaced classes from the output. */
+  public static final String STRATEGY_OMIT = "omit";
+
+  /**
+   * Strategy which renames misplaced classes to their "proper" location based on their
+   * fully-qualified class name.
+   */
+  public static final String STRATEGY_MOVE = "move";
+
+  private static MisplacedClassProcessorFactory me;
+
+  public static synchronized MisplacedClassProcessorFactory getInstance() {
+    if (me == null) {
+      me = new MisplacedClassProcessorFactory();
+    }
+    return me;
+  }
+
+  private MisplacedClassProcessorFactory() {}
+
+  /**
+   * Returns the default misplaced class processor, which is "omit".
+   *
+   * @param context The processing context for the processor.
+   * @return
+   */
+  public MisplacedClassProcessor getDefaultProcessor(ProcessingContext context) {
+    return new OmitMisplacedClassProcessor(context);
+  }
+
+  /**
+   * Creates a MisplacedClassProcessor according for the given strategy name with the given context.
+   *
+   * @param name The case-insensitive user-level strategy name (see the STRATEGY_* constants).
+   * @param context The ProcessingContext to pass to the processor.
+   * @return The MisplacedClassProcessor corresponding to the strategy name, or the result of
+   * getDefaultProcessor(context) if name is null.
+   * @throws IllegalAccessException if an unrecognized non-null strategy name is specified.
+   */
+  public MisplacedClassProcessor getProcessorForName(String name, ProcessingContext context) {
+    if (name == null) {
+      return getDefaultProcessor(context);
+    }
+
+    name = name.toLowerCase();
+    // NB(gmalmquist): Have to support java6, so no string switch statements allowed.
```

Comment:
It doesn't bother me that much -- I just wanted to explain why it's set up like this.